src/SBInclinedExponential.cpp

```diff
@@ -220,15 +220,11 @@ namespace galsim {
 
     double SBInclinedExponential::SBInclinedExponentialImpl::maxSB() const
     {
-        // When the disk is face on, the max SB is flux / 2 pi r0^2
-        // When the disk is edge on, the max SB is flux / 2 pi r0^2 * (r0/h0)
-        double maxsb = _flux * _inv_r0 * _inv_r0 / (2. * M_PI);
-        // The relationship for inclinations in between these is not linear.
-        // Empirically, it is vaguely linearish in sqrt(cosi), so we use that for
-        // the interpolation.  It's accurate to ~10-20% for moderate values of h0/r0.
-        double sc = sqrt(std::abs(_cosi));
-        maxsb *= (_h0 * sc + _r0 * (1.-sc)) / _h0;
-        return std::abs(maxsb);
+        // Maximum surface brightness isn't analytic, and is only needed for photon-shooting,
+        // so this is disabled.
+        throw std::runtime_error(
+            "The maxSB method of SBInclinedExponential is not yet implemented");
+        return 0;
```

Comment:
No need for a warning.  The documentation about it already warns the user that it is not always terribly accurate:
```
        Some profiles will return the exact peak SB, typically equal to the value of
        obj.xValue(obj.centroid()).  However, not all profiles (e.g. Convolution) know how to
        calculate this value without just drawing the image and checking what the maximum value is.
        Clearly, this would be inefficient, so in these cases, some kind of estimate is returned,
        which will generally be conservative on the high side.

        This routine is mainly used by the photon shooting process, where an overestimate of
        the maximum surface brightness is acceptable.
```